numpy/f2py/cfuncs.py

```diff
@@ -1152,6 +1182,31 @@
 """
 
 
+cfuncs['try_pyarr_from_character'] = """\
+static int try_pyarr_from_character(PyObject* obj, character* v) {
+    PyArrayObject *arr = (PyArrayObject*)obj;
+    if (!obj) return -2;
```

Comment:
In other parts of the code, -2 seems to be returned for both shape mismatches and issues with contiguous memory, perhaps a comment would be good?

Reply:
Returning -2 is consistent with `TRYPYARRAYTEMPLATE`. That said, I think all `try_pyarr_from_*` functions require a revision as their return values are not used other than checking for a non-zero status. In fact, these functions never fail.. (notice the `try_` prefix). So, I would leave addressing this suggestion when revising the `try_..` support in general.